### PR TITLE
GS: Improve optimizing scissoring texture when REPEAT sampling

### PR DESCRIPTION
**Pull request by refractionpcsx2**
### Description of Changes
Improves the handling of repeat range handling when the draw is scissored.

### Rationale behind Changes
previously it went off the full range of the read to see if it looped, not just the scissored area, so a lot of the time this check was failing.  A working example of this was with the Godfather, where it ended up expanding a target from 512x256 to 512x512, even though it was scissored, this meant later on a tex in rt check was picking up the wrong target.

### Suggested Testing Steps
Smoke test mostly, and The Godfather, make sure garbage isn't appearing at the bottom when moving (in vulkan)

### CODE REVIEW

File: pcsx2/GS/Renderers/HW/GSRendererHW.cpp [hunk 1]
```diff
@@ -5709,7 +5710,7 @@ GSRendererHW::CLUTDrawTestResult GSRendererHW::PossibleCLUTDraw()
 	if (m_process_texture)
 	{
 		// If we're using a texture to draw our CLUT/whatever, we need the GPU to write back dirty data we need.
-		const GSVector4i r = GetTextureMinMax(m_cached_ctx.TEX0, m_cached_ctx.CLAMP, m_vt.IsLinear(), false).coverage;
+		const GSVector4i r = GetTextureMinMax(m_cached_ctx.TEX0, m_cached_ctx.CLAMP, m_vt.IsLinear(), false, m_vt.m_primclass == GS_SPRITE_CLASS && PrimitiveCoversWithoutGaps()).coverage;
```
Comment (stenzek) [inline pcsx2/GS/Renderers/HW/GSRendererHW.cpp:5713]: I think it's worth moving `PrimitiveCoversWithoutGaps()` into `GSState` (if it isn't already), and then moving this call into `GetTextureMinMax()`. It's a potentially costly operation (that's why I memoized the result), and we don't want to be doing it for all sprites unless all the preconditions for needing it are satisfied.

Comment (refractionpcsx2) [reply pcsx2/GS/Renderers/HW/GSRendererHW.cpp:5713]: I did think about doing that, yeah, I may very well do that.  I still only do it when it's FST and sprite and no gaps, there's no other scenario

Comment (stenzek) [reply pcsx2/GS/Renderers/HW/GSRendererHW.cpp:5713]: Maybe we should consider equal-q (or q=1).. IIRC the expanding case I was looking at the other day used normalized coordinates.

Comment (refractionpcsx2) [reply pcsx2/GS/Renderers/HW/GSRendererHW.cpp:5713]: we could do, yea, since the vertex trace has already worked out the texture dimensions

**Review by Mrlinkwii — APPROVED** [review]

**Review by kamfretoz — APPROVED** [review]

**Pull request merged**